### PR TITLE
Refactor code, remove unnecessary middleware, add comments

### PR DESCRIPTION
**Pull request by z3ht**

### CODE REVIEW

File: python_scraper/python_scraper/spiders/sec_nport_p.py [hunk 1]
```diff
@@ -6,25 +6,43 @@
...
                 'Accept-Language': 'en',
             })
 
+    # Lambda method added to Scrapy parsing queue
+    # Called in the parse lambda
+    #
+    # Response contains parsed XML nport info
     def parse_xml(self, response):
         response.selector.register_namespace('x', 'http://www.sec.gov/edgar/nport')
-        yield CompanyItem(
```
Comment (nicolls1) [inline python_scraper/python_scraper/spiders/sec_nport_p.py]: Why not yield?

File: python_scraper/python_scraper/pipelines.py [hunk 2]
```diff
@@ -12,13 +12,23 @@
...
-            if isinstance(item, CompanyItem) and Company.objects.filter(series_lei=instance.series_lei,
-                                                                        rep_pd_date=instance.rep_pd_date).exists():
-                # skip items already in database, define series_lei and rep_pd_date to be unique
-                commit = False
-            item.save(commit=commit)
+        # Commit if:
+        #   Item is a Django Item
+        #   Item is not currently in the database
+        #       - series_lei and rep_pd_date act as primary keys
+        item.save(
```
Comment (nicolls1) [inline python_scraper/python_scraper/pipelines.py]: I think you are right with getting rid of the second if statement. The thing is we don't want item.save() to be called when its not a DjangoItem, other items could be included here as well, so I think we need the first one.

**Comment by nicolls1** [comment]
Thanks! Just a couple comments

File: python_scraper/python_scraper/spiders/sec_nport_p.py (hunk 1 above)
Comment (z3ht) [reply python_scraper/python_scraper/spiders/sec_nport_p.py]: Either works. The return keyword seems more intuitive to its use.

Comment (nicolls1) [reply python_scraper/python_scraper/spiders/sec_nport_p.py]: Okie, sure we can change for this case. Maybe you know, but if you don't, you can only return one thing from a function but you can yield as many as you like. In the future we will most likely be yielding multiple items from the xml so then we would yield them.

File: python_scraper/python_scraper/pipelines.py (hunk 2 above)
Comment (z3ht) [reply python_scraper/python_scraper/pipelines.py]: My bad on this. Its been fixed now

File: python_scraper/python_scraper/spiders/sec_nport_p.py (hunk 1 above)
Comment (z3ht) [reply python_scraper/python_scraper/spiders/sec_nport_p.py]: Yes I'm aware of the difference. When we're returning from a for loop it makes sense to yield but in this case we're not so return seems best.

**Comment by z3ht** [comment]
Alright the PR should be ready

**Pull request merged**